packages/ui-components/src/chat/chat.tsx

```diff
@@ -0,0 +1,228 @@
+ */
+function getFormattedTimestamp(): string {
+    const now = new Date();
+
+    const hours = now.getHours().toString().padStart(2, '0');
+    const minutes = now.getMinutes().toString().padStart(2, '0');
+    const day = now.getDate().toString().padStart(2, '0');
+    const month = (now.getMonth() + 1).toString().padStart(2, '0'); // +1 because months are 0-indexed
+    const year = now.getFullYear();
+
+    return `${hours}:${minutes} ${day}/${month}/${year}`;
+}
+
+/**
+ * A function to scroll to the bottom of the chat so that the latest message is visible
+ */
+function scrollToBottom(): void {
+    setTimeout(() => {
+        const scrollContainer = document.getElementById('scrollContainer');
+        if (scrollContainer) {
+            scrollContainer.scrollTop = scrollContainer.scrollHeight;
+        }
+    }, 100); // Timeout of 100ms to allow the DOM to update with latest values
+}
+
+/**
+ * A chat component
+ *
+ * @param {ChatProps} props - the component props
+ */
+function Chat({ ...props }: ChatProps): JSX.Element {
+    const [reply, setReply] = React.useState('');
+
+    const [messages, setMessages] = React.useState(props.value ?? []);
+
+    const onSubmitMessage = (): void => {
+        if (reply) {
+            // Create a new message
+            const newMessage = {
+                id: uuidv4(),
+                message: reply,
+                timestamp: getFormattedTimestamp(),
+            };
+            const newMessages = [...messages, newMessage];
+
+            // Add the new message to the chat
+            if (props.value !== undefined) {
+                props.onAdd?.(newMessages);
+            } else {
+                setMessages(newMessages);
+            }
+
+            // Clear the reply field and scroll to the bottom of the chat to show latest message
+            setReply('');
+            scrollToBottom();
+        }
+    };
+
+    const onEditMessage = (message: Message): void => {
+        // Find the message to edit and replace it with the new message
+        const newMessages = messages.map((m) => {
+            if (m.id === message.id) {
+                return message;
+            }
+            return m;
+        });
+        // Update the chat
+        if (props.value !== undefined) {
+            props.onEdit?.(newMessages);
+        } else {
+            setMessages(newMessages);
+        }
+    };
+
+    const onDeleteMessage = (id: string): void => {
+        // Remove the message with the given id
+        const newMessages = messages.filter((message) => message.id !== id);
+        // Update the chat
+        if (props.value !== undefined) {
+            props.onDelete?.(id);
+        } else {
+            setMessages(newMessages);
+        }
+    };
+
+    React.useEffect(() => {
+        // Scroll to the bottom of the chat
+        scrollToBottom();
+    }, []);
+
+    React.useEffect(() => {
+        setMessages(props.value ?? []);
+    }, [props.value]);
```

Comment:
Could these maybe be combined and always scroll to bottom when new messages appear?

```suggestion
    React.useEffect(() => {    
        setMessages(props.value ?? []);
        // TODO: might need to be debounced
        scrollToBottom();
    }, [props.value]);
```

Reply:
Resolved in person